--only-smart-pointers should enable webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor

rniwa opened the pull request:
#### 438692dd840d555018cb25e55ae5bc1dc177a7e4
<pre>
--only-smart-pointers should enable webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor
<a href="https://bugs.webkit.org/show_bug.cgi?id=273674">https://bugs.webkit.org/show_bug.cgi?id=273674</a>

Reviewed by Geoffrey Garen and David Kilzer.

This PR enables two WebKit checkers webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor
with --only-smart-pointers.

* Tools/Scripts/build-and-analyze:
(make_analyzer_flags):

Canonical link: <a href="https://commits.webkit.org/278352@main">https://commits.webkit.org/278352@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/42521ff77a9ae0b6013ed78d2946b3b476cab26a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50257 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29550 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2554 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53516 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/947 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52560 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35778 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/585 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41006 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52356 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27213 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43257 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22105 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/24631 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/509 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8635 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/46617 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/572 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55101 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25353 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48408 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26616 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43441 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47431 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27478 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7269 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26346 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me

ddkilzer commented on Tools/Scripts/build-and-analyze:
```diff
@@ -80,6 +80,8 @@ def make_analyzer_flags(output_dir, args):
+            'webkit.NoUncountedMemberChecker',
+            'webkit.RefCntblBaseVirtualDtor',
             'alpha.webkit.UncountedCallArgsChecker',
             'alpha.webkit.UncountedLocalVarsChecker',
```
Super nit:  I'd keep these in alphabetical order, but it's okay to land without doing that.

rniwa replied:
ok, I can report them.

ddkilzer reviewed (approved):
r=me (nit is not required to fix in order to land this)

rniwa committed:
--only-smart-pointers should enable webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor
https://bugs.webkit.org/show_bug.cgi?id=273674

Reviewed by Geoffrey Garen and David Kilzer.

This PR enables two WebKit checkers webkit.NoUncountedMemberChecker and webkit.RefCntblBaseVirtualDtor
with --only-smart-pointers.

* Tools/Scripts/build-and-analyze:
(make_analyzer_flags):

Canonical link: https://commits.webkit.org/278352@main
